src/net_processing.cpp

```diff
@@ -3993,6 +4028,22 @@ void PeerManager::ProcessMessage(CNode& pfrom, const std::string& msg_type, CDat
         return;
     }
 
+    std::chrono::microseconds current_time = GetTime<std::chrono::microseconds>();
+
+    // Record an (expected) reconciliation request with parameters to respond when time comes.
+    // All initial reconciliation responses will be done at the same time to prevent tx-related privacy leaks.
+    if (msg_type == NetMsgType::REQRECON) {
+        if (peer->m_recon_state == nullptr) return;
+        if (!peer->m_recon_state->m_sender) return;
+        if (peer->m_recon_state->m_incoming_recon != Peer::ReconState::ReconPhase::NONE) return;
+        uint16_t peer_recon_set_size, peer_q;
+        vRecv >> peer_recon_set_size >> peer_q;
+        peer->m_recon_state->m_incoming_recon = Peer::ReconState::ReconPhase::INIT_REQUESTED;
+        peer->m_recon_state->m_remote_set_size = peer_recon_set_size;
+        peer->m_recon_state->m_remote_q = double(peer_q / Q_PRECISION);
+        peer->m_recon_state->m_next_recon_respond = NextReconRespond(current_time);
```

Comment:
I think there's no real difference with this design though?
We gain privacy by not allowing 2 peers to monitor our mempool, because we respond to them at the same time (every 2 seconds).

Snapshoting earlier would hide the transactions received during the last 2 seconds and disable this kind of monitoring, but then reconciliations would not be so "fresh" (txs received during last 2s go into the next one).
Not sure this kind of monitoring is that dangerous, especially since we *also* have a Poisson on adding to recon sets.